src/controllers/simplified-account/services/payment-links/create/payment-link-amount.controller.ts

```diff
@@ -0,0 +1,101 @@
+import paths from '@root/paths'
+import formatServiceAndAccountPathsFor from '@utils/simplified-account/format/format-service-and-account-paths-for'
+import { ServiceRequest, ServiceResponse } from '@utils/types/express'
+import { response } from '@utils/response'
+import { CREATE_SESSION_KEY, PaymentLinkCreationSession } from './constants'
+import lodash from 'lodash'
+import { paymentLinkSchema } from '@utils/simplified-account/validation/payment-link.schema'
+import { validationResult } from 'express-validator'
+import formatValidationErrors from '@utils/simplified-account/format/format-validation-errors'
+import { penceToPounds, safeConvertPoundsStringToPence } from '@utils/currency-formatter'
+
+function getSession<T>(req: ServiceRequest<T>, res: ServiceResponse) {
+  const { service, account } = req
+  const session = lodash.get(req, CREATE_SESSION_KEY, {} as PaymentLinkCreationSession)
+  if (lodash.isEmpty(session)) {
+    res.redirect(formatServiceAndAccountPathsFor(paths.simplifiedAccount.paymentLinks.index, service.externalId, account.type))
```

Comment:
I'd recommend hoisting this out into the controller methods so you can return in the controller on redirect. this will avoid any other parts of the controller methods being executed one the redirect is triggered